app/javascript/bundles/Groceries/components/Groceries.tsx

```diff
@@ -89,11 +64,28 @@ export default class Groceries extends React.Component<RouteComponentProps, Groc
         throw new Error("Network response was not ok.");
       })
       .then(response => this.setState(
-        { itemsInList: response.map(groceryListItem => ({ ...groceryListItem, quantity: 1 })) }
+        { groceryItems: response.map(groceryListItem => ({ ...groceryListItem, quantity: 1 })) }
       ))
       .catch(() => this.props.history.push("/"));
   }
 
+  onSearch(text: string): void {
+    const cartLineItems = this.state.groceryLineItems.filter(x => x.location == GroceryLineItemLocation.Cart)
```

Comment:
One downside to combining the two... We can think about it and re-optimize. For instance, we could have two separate lists in the state, but just be careful with our callbacks to keep the logic separated.